src/models/test/ThemeProvider.test.js

```diff
@@ -89,14 +87,13 @@ describe('ThemeProvider', () => {
     // Setup Child
     class Child extends React.Component {
       componentWillMount() {
-        this.context[CHANNEL](theme => {
-          // eslint-disable-next-line react/prop-types
-          expect(theme).toEqual(themes[this.props.shouldHaveTheme])
-          childRendered++ // eslint-disable-line no-plusplus
-          if (childRendered === Object.keys(themes).length) {
-            done()
-          }
-        })
+        const theme = this.context[CHANNEL].getState()
```

Comment:
I'm not familiar with `theming`, does this imply there would be a similar context type change as discussed in https://github.com/styled-components/styled-components/pull/1048#discussion_r130274738?